ethcore/src/verification/verification.rs

```diff
@@ -329,11 +335,14 @@ fn verify_parent(header: &Header, parent: &Header, engine: &dyn Engine) -> Resul
 		return Err(BlockError::RidiculousNumber(OutOfBounds { min: Some(1), max: None, found: header.number() }).into());
 	}
 
-	let parent_gas_limit = *parent.gas_limit();
-	let min_gas = parent_gas_limit - parent_gas_limit / gas_limit_divisor;
-	let max_gas = parent_gas_limit + parent_gas_limit / gas_limit_divisor;
-	if header.gas_limit() <= &min_gas || header.gas_limit() >= &max_gas {
-		return Err(From::from(BlockError::InvalidGasLimit(OutOfBounds { min: Some(min_gas), max: Some(max_gas), found: *header.gas_limit() })));
+	if engine.gas_limit_override(header).is_none() {
```

Comment:
That's pretty wasteful to call the contract twice during verification (one for `verify_header` and second time here), why not move the other gas_limit verification in here?